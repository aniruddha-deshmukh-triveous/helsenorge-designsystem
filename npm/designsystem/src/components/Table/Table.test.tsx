import React from 'react';
import { screen, render } from '@testing-library/react';
import { SmallViewportVariant, Table, TableBody, TableHead, TableHeadCell, TableCell, TableRow, HeaderCategory } from './';

describe('Gitt at Tabel skal vises', (): void => {
  describe('Når den skal vises', (): void => {
    it('Så vises en tabell', (): void => {
      const { container } = render(
        <Table smallViewportVariant={SmallViewportVariant.horizontalscroll} testId="test01">
          <TableHead category={HeaderCategory.normal}>
            <TableRow key="0">
              <TableHeadCell>Navn</TableHeadCell>
              <TableHeadCell>Beskrivelse</TableHeadCell>
            </TableRow>
          </TableHead>
          <TableBody>
            <TableRow key="1">
              <TableCell dataLabel="Navn">Hans Nilsen</TableCell>
              <TableCell dataLabel="Beskrivelse">En ganske lang beskrivelse...</TableCell>
            </TableRow>
          </TableBody>
        </Table>
      );

      const table = screen.getByTestId('test01');
      const child = screen.getByText('Hans Nilsen');

      expect(table).toHaveClass('table');
      expect(child).toHaveClass('table__cell');

      expect(container).toMatchSnapshot();
    });
  });
});
