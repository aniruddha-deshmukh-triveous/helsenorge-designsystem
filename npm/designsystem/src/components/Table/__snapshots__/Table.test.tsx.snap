// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gitt at Table skal vises Når den skal vises Så vises en tabell 1`] = `
<div>
  <table
    class="table"
    data-analyticsid="table"
    data-testid="test01"
  >
    <thead
      class="table__head table__head--normal"
    >
      <tr
        class="table-row"
      >
        <th
          class="table__head-cell"
          scope="col"
        >
          Navn
        </th>
        <th
          class="table__head-cell"
          scope="col"
        >
          Beskrivelse
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="table-row"
      >
        <td
          class="table__cell"
          data-label="Navn"
        >
          Hans Nilsen
        </td>
        <td
          class="table__cell"
          data-label="Beskrivelse"
        >
          En ganske lang beskrivelse...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Når den skal vises med blokk-visning på md skjerm Når breakpoint er lg Så vises en tabell uten responsiv visning 1`] = `
<div>
  <table
    class="table"
    data-analyticsid="table"
    data-testid="test-table"
  >
    <thead
      class="table__head table__head--normal"
    >
      <tr
        class="table-row"
      >
        <th
          class="table__head-cell"
          scope="col"
        >
          Navn
        </th>
        <th
          class="table__head-cell"
          scope="col"
        >
          Beskrivelse
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="table-row"
      >
        <td
          class="table__cell"
          data-label="Navn"
        >
          Hans Nilsen
        </td>
        <td
          class="table__cell"
          data-label="Beskrivelse"
        >
          En ganske lang beskrivelse...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Når den skal vises med blokk-visning på md skjerm Når breakpoint er md Så vises en tabell 1`] = `
<div>
  <table
    class="table table--block-md"
    data-analyticsid="table"
    data-testid="test-table"
  >
    <thead
      class="table__head table__head--normal"
    >
      <tr
        class="table-row"
      >
        <th
          class="table__head-cell"
          scope="col"
        >
          Navn
        </th>
        <th
          class="table__head-cell"
          scope="col"
        >
          Beskrivelse
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="table-row"
      >
        <td
          class="table__cell"
          data-label="Navn"
        >
          Hans Nilsen
        </td>
        <td
          class="table__cell"
          data-label="Beskrivelse"
        >
          En ganske lang beskrivelse...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Når den skal vises med centeredoverflow-visning på lg skjerm Når breakpoint er lg Når tabell er smalere enn skjermen Så vises en tabell med responsiv visning 1`] = `
<div>
  <table
    class="table table--centeredoverflow-lg"
    data-analyticsid="table"
    data-testid="test-table"
  >
    <thead
      class="table__head table__head--normal"
    >
      <tr
        class="table-row"
      >
        <th
          class="table__head-cell"
          scope="col"
        >
          Navn
        </th>
        <th
          class="table__head-cell"
          scope="col"
        >
          Beskrivelse
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="table-row"
      >
        <td
          class="table__cell"
          data-label="Navn"
        >
          Hans Nilsen
        </td>
        <td
          class="table__cell"
          data-label="Beskrivelse"
        >
          En ganske lang beskrivelse...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Når den skal vises med centeredoverflow-visning på lg skjerm Når fallback er horizontalscroll Når touch ikke er støttet Så vises en tabell uten responsiv visning 1`] = `
<div>
  <table
    class="table"
    data-analyticsid="table"
    data-testid="test-table"
  >
    <thead
      class="table__head table__head--normal"
    >
      <tr
        class="table-row"
      >
        <th
          class="table__head-cell"
          scope="col"
        >
          Navn
        </th>
        <th
          class="table__head-cell"
          scope="col"
        >
          Beskrivelse
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="table-row"
      >
        <td
          class="table__cell"
          data-label="Navn"
        >
          Hans Nilsen
        </td>
        <td
          class="table__cell"
          data-label="Beskrivelse"
        >
          En ganske lang beskrivelse...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Når den skal vises med centeredoverflow-visning på lg skjerm Når tabell er bredere enn skjermen Når fallback er block Så vises en tabell med responsiv visning 1`] = `
<div>
  <table
    class="table table--block-lg"
    data-analyticsid="table"
    data-testid="test-table"
  >
    <thead
      class="table__head table__head--normal"
    >
      <tr
        class="table-row"
      >
        <th
          class="table__head-cell"
          scope="col"
        >
          Navn
        </th>
        <th
          class="table__head-cell"
          scope="col"
        >
          Beskrivelse
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="table-row"
      >
        <td
          class="table__cell"
          data-label="Navn"
        >
          Hans Nilsen
        </td>
        <td
          class="table__cell"
          data-label="Beskrivelse"
        >
          En ganske lang beskrivelse...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Når den skal vises med centeredoverflow-visning på lg skjerm Når tabell er bredere enn skjermen Når ingen fallback er definert Så vises en tabell uten responsiv visning 1`] = `
<div>
  <table
    class="table"
    data-analyticsid="table"
    data-testid="test-table"
  >
    <thead
      class="table__head table__head--normal"
    >
      <tr
        class="table-row"
      >
        <th
          class="table__head-cell"
          scope="col"
        >
          Navn
        </th>
        <th
          class="table__head-cell"
          scope="col"
        >
          Beskrivelse
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="table-row"
      >
        <td
          class="table__cell"
          data-label="Navn"
        >
          Hans Nilsen
        </td>
        <td
          class="table__cell"
          data-label="Beskrivelse"
        >
          En ganske lang beskrivelse...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Når den skal vises med horizontalscroll-visning på md skjerm Når touch er støttet Når breakpoint er lg Så vises en tabell uten responsiv visning 1`] = `
<div>
  <table
    class="table"
    data-analyticsid="table"
    data-testid="test-table"
  >
    <thead
      class="table__head table__head--normal"
    >
      <tr
        class="table-row"
      >
        <th
          class="table__head-cell"
          scope="col"
        >
          Navn
        </th>
        <th
          class="table__head-cell"
          scope="col"
        >
          Beskrivelse
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="table-row"
      >
        <td
          class="table__cell"
          data-label="Navn"
        >
          Hans Nilsen
        </td>
        <td
          class="table__cell"
          data-label="Beskrivelse"
        >
          En ganske lang beskrivelse...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Når den skal vises med horizontalscroll-visning på md skjerm Når touch er støttet Når breakpoint er md Så vises en tabell med responsiv visning 1`] = `
<div>
  <div
    class="horizontalscroll"
    data-testid="horizontal-scroll"
  >
    <div
      class="horizontalscroll__viewport"
      tabindex="0"
    >
      <div
        class="horizontalscroll__indicator horizontalscroll__indicator--left"
      />
      <div
        class="horizontalscroll__indicator horizontalscroll__indicator--right"
      />
      <div />
      <table
        class="table"
        data-analyticsid="table"
        data-testid="test-table"
      >
        <thead
          class="table__head table__head--normal"
        >
          <tr
            class="table-row"
          >
            <th
              class="table__head-cell"
              scope="col"
            >
              Navn
            </th>
            <th
              class="table__head-cell"
              scope="col"
            >
              Beskrivelse
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="table-row"
          >
            <td
              class="table__cell"
              data-label="Navn"
            >
              Hans Nilsen
            </td>
            <td
              class="table__cell"
              data-label="Beskrivelse"
            >
              En ganske lang beskrivelse...
            </td>
          </tr>
        </tbody>
      </table>
      <div />
    </div>
  </div>
</div>
`;

exports[`Når den skal vises med horizontalscroll-visning på md skjerm Når touch ikke er støttet Når fallback er block Så vises en tabell med responsiv visning 1`] = `
<div>
  <table
    class="table table--block-md"
    data-analyticsid="table"
    data-testid="test-table"
  >
    <thead
      class="table__head table__head--normal"
    >
      <tr
        class="table-row"
      >
        <th
          class="table__head-cell"
          scope="col"
        >
          Navn
        </th>
        <th
          class="table__head-cell"
          scope="col"
        >
          Beskrivelse
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="table-row"
      >
        <td
          class="table__cell"
          data-label="Navn"
        >
          Hans Nilsen
        </td>
        <td
          class="table__cell"
          data-label="Beskrivelse"
        >
          En ganske lang beskrivelse...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Når den skal vises med horizontalscroll-visning på md skjerm Når touch ikke er støttet Når fallback er centeredoverflow Når tabell er bredere enn skjermen Så vises en tabell uten responsiv visning 1`] = `
<div>
  <table
    class="table"
    data-analyticsid="table"
    data-testid="test-table"
  >
    <thead
      class="table__head table__head--normal"
    >
      <tr
        class="table-row"
      >
        <th
          class="table__head-cell"
          scope="col"
        >
          Navn
        </th>
        <th
          class="table__head-cell"
          scope="col"
        >
          Beskrivelse
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="table-row"
      >
        <td
          class="table__cell"
          data-label="Navn"
        >
          Hans Nilsen
        </td>
        <td
          class="table__cell"
          data-label="Beskrivelse"
        >
          En ganske lang beskrivelse...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Når den skal vises med horizontalscroll-visning på md skjerm Når touch ikke er støttet Når ingen fallback er definert Så vises en tabell uten responsiv visning 1`] = `
<div>
  <table
    class="table"
    data-analyticsid="table"
    data-testid="test-table"
  >
    <thead
      class="table__head table__head--normal"
    >
      <tr
        class="table-row"
      >
        <th
          class="table__head-cell"
          scope="col"
        >
          Navn
        </th>
        <th
          class="table__head-cell"
          scope="col"
        >
          Beskrivelse
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="table-row"
      >
        <td
          class="table__cell"
          data-label="Navn"
        >
          Hans Nilsen
        </td>
        <td
          class="table__cell"
          data-label="Beskrivelse"
        >
          En ganske lang beskrivelse...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
